tests/integration/deploy/test_managed_stack_deploy.py

```diff
@@ -0,0 +1,120 @@
+from parameterized import parameterized
+
+from samcli.lib.bootstrap.bootstrap import SAM_CLI_STACK_NAME
+from samcli.lib.config.samconfig import DEFAULT_CONFIG_FILE_NAME
+from tests.integration.deploy.deploy_integ_base import DeployIntegBase
+from tests.integration.package.package_integ_base import PackageIntegBase
+from tests.testing_utils import RUNNING_ON_CI, RUNNING_TEST_FOR_MASTER_ON_CI, RUN_BY_CANARY
+from tests.testing_utils import run_command, run_command_with_input
+
+# Deploy tests require credentials and CI/CD will only add credentials to the env if the PR is from the same repo.
+# This is to restrict package tests to run outside of CI/CD, when the branch is not master or tests are not run by Canary
+SKIP_DEPLOY_TESTS = RUNNING_ON_CI and RUNNING_TEST_FOR_MASTER_ON_CI and not RUN_BY_CANARY
+CFN_SLEEP = 3
+TIMEOUT = 300
+CFN_PYTHON_VERSION_SUFFIX = os.environ.get("PYTHON_VERSION", "0.0.0").replace(".", "-")
+
+
+@skipIf(SKIP_DEPLOY_TESTS, "Skip deploy tests in CI/CD only")
+class TestManagedStackDeploy(PackageIntegBase, DeployIntegBase):
+    @classmethod
+    def setUpClass(cls):
+        PackageIntegBase.setUpClass()
+        DeployIntegBase.setUpClass()
+
+    def setUp(self):
+        self.cfn_client = boto3.client("cloudformation")
+        self.s3_client = boto3.client("s3")
+        self.sns_arn = os.environ.get("AWS_SNS")
+        self.stacks = []
+        time.sleep(CFN_SLEEP)
+        super().setUp()
+
+    def tearDown(self):
+        shutil.rmtree(os.path.join(os.getcwd(), ".aws-sam", "build"), ignore_errors=True)
+        for stack in self.stacks:
+            # because of the termination protection, do not delete aws-sam-cli-managed-default stack
+            stack_name = stack["name"]
+            if stack_name != SAM_CLI_STACK_NAME:
+                region = stack.get("region")
+                cfn_client = (
+                    self.cfn_client if not region else boto3.client("cloudformation", config=Config(region_name=region))
+                )
+                cfn_client.delete_stack(StackName=stack_name)
+        super().tearDown()
+
+    @parameterized.expand(["aws-serverless-function.yaml"])
+    def test_managed_stack_creation(self, template_file):
+        self._delete_managed_stack(self.cfn_client, self.s3_client)
+        self.assertFalse(self._does_stack_exist(self.cfn_client, SAM_CLI_STACK_NAME))
+
+        template_path = self.test_data_path.joinpath(template_file)
+
+        stack_name = self._method_to_stack_name(self.id())
+        self.stacks.append({"name": stack_name})
+
+        deploy_command_list = self.get_deploy_command_list(
+            template_file=template_path,
+            stack_name=stack_name,
+            capabilities="CAPABILITY_IAM",
+            force_upload=True,
+            notification_arns=self.sns_arn,
+            parameter_overrides="Parameter=Clarity",
+            kms_key_id=self.kms_key,
+            tags="integ=true clarity=yes foo_bar=baz",
+            resolve_s3=True,
```

Comment:
should we go through the deploy --guided workflow too? since both the interactive and non interactive model can create the managed stack.